### PR TITLE
User references

### LINKED ISSUE #420: User page references/reference list
From beta ui, with tabs to sort from hosts/guests/etc

### PR DESCRIPTION
**Pull request by darrenvong**
Covers what's needed in #420 except the filters (which has just been added recently...). Think I'll do that in a separate/follow-up PR.

### CODE REVIEW

File: app/frontend/src/features/userPage/UserPage.tsx [hunk 1]
```diff
@@ -1,11 +1,13 @@
-import { Grid, makeStyles } from "@material-ui/core";
-import React from "react";
+import { Grid, makeStyles, Typography } from "@material-ui/core";
```
Comment (darrenvong) [inline app/frontend/src/features/userPage/UserPage.tsx]: This file is more of a placeholder to keep everything in this branch working, will likely be deleted before this gets merged due to Andrea's work.

File: app/frontend/src/components/Pill/Pill.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import classNames from "classnames";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    backgroundColor: theme.palette.secondary.main,
+    padding: theme.spacing(0.5, 1),
+    textAlign: "center",
+  },
+  rounded: {
+    borderRadius: theme.typography.pxToRem(20),
```
Comment (lucaslcode) [inline app/frontend/src/components/Pill/Pill.tsx]:
```suggestion
    borderRadius: theme.shape.borderRadius * 2,
```

File: app/frontend/src/components/TabBar/TabBar.tsx [hunk 3]
```diff
@@ -3,9 +3,7 @@ import { makeStyles } from "@material-ui/core/styles";
 import React from "react";
 
 const useStyles = makeStyles((theme) => ({
-  tabRoot: {
-    minWidth: theme.spacing(25),
-  },
+  tabRoot: {},
```
Comment (lucaslcode) [inline app/frontend/src/components/TabBar/TabBar.tsx]:
```suggestion
```

File: app/frontend/src/service/user.ts [hunk 4]
```diff
@@ -305,3 +309,46 @@ export async function completeSignup({
...
+  userId,
+  offset,
+  referenceType,
+}: GetReferencesInput) {
+  const req = new GetGivenReferencesReq();
+  req.setFromUserId(userId);
+  if (referenceType) {
+    req.setTypeFilter(new OptionalReferenceType().setValue(referenceType));
+  }
+  req.setStartAt(offset);
```
Comment (lucaslcode) [inline app/frontend/src/service/user.ts:331]: You can tell this part of the api is much older than the community stuff 😆

**Review by lucaslcode — DISMISSED** [review]
Very minor stuff

File: app/frontend/src/components/HostingStatus/HostingStatus.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+import { hostingStatusLabels } from "../../features/profile/constants";
+import { HostingStatus as THostingStatus } from "../../pb/api_pb";
+import { CouchIcon } from "../Icons";
+import IconText from "../IconText";
+
+const useStyles = makeStyles({
+  hostingAbilityContainer: {
+    display: "flex",
+    alignItems: "center",
```
Comment (andreaowu) [inline app/frontend/src/components/HostingStatus/HostingStatus.tsx]: nit: alphabetize these?

File: app/frontend/src/components/TabBar/TabBar.tsx (hunk 3 above)
Comment (andreaowu) [reply app/frontend/src/components/TabBar/TabBar.tsx]: delete `useStyles` altogether?

File: app/frontend/src/components/UserSummary.tsx [hunk 6]
```diff
@@ -8,34 +8,28 @@ import { Skeleton } from "@material-ui/lab";
...
 import { User } from "../pb/api_pb";
 import Avatar from "./Avatar";
 import ScoreBar from "./Bar/ScoreBar";
-import { CouchIcon } from "./Icons";
-import IconText from "./IconText";
 
-const useStyles = makeStyles((theme) => ({
-  root: { display: "flex", alignItems: "center", padding: 0 },
+export const useStyles = makeStyles((theme) => ({
+  root: { display: "flex", alignItems: "center", padding: 0, width: "100%" },
```
Comment (andreaowu) [inline app/frontend/src/components/UserSummary.tsx]: nit: formatting? put a comma after the `width: "100%"` + alphabetize

File: app/frontend/src/components/UserSummary.tsx [hunk 7]
```diff
@@ -8,34 +8,28 @@ import { Skeleton } from "@material-ui/lab";
...
-  hostingAbilityContainer: {
-    display: "flex",
-    alignItems: "center",
-  },
   avatar: {
-    width: theme.spacing(7),
-    height: theme.spacing(7),
+    width: theme.spacing(9),
+    height: theme.spacing(9),
     marginInlineEnd: theme.spacing(2),
```
Comment (andreaowu) [inline app/frontend/src/components/UserSummary.tsx]: nit: alphabetize

File: app/frontend/src/features/userPage/UserReferences.tsx [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+  },
+  listItem: {
+    flexDirection: "column",
+    alignItems: "flex-start",
+    "& > * + *": {
+      marginBlockStart: theme.spacing(2),
+    },
+    borderBlockEnd: `${theme.typography.pxToRem(1)} solid ${
+      theme.palette.grey[300]
+    }`,
```
Comment (andreaowu) [inline app/frontend/src/features/userPage/UserReferences.tsx]: nit: alphabetize

File: app/frontend/src/features/userPage/UserSection.tsx [hunk 9]
```diff
@@ -1,17 +1,17 @@
...
-    borderRadius: theme.shape.borderRadius,
-    marginBottom: theme.spacing(2),
+    paddingBlockStart: theme.spacing(1),
+    paddingBlockEnd: theme.spacing(1),
+    borderBlockStart: `${theme.typography.pxToRem(1)} solid ${
+      theme.palette.grey[300]
+    }`,
+    "& > * + *": {
+      marginBlockStart: theme.spacing(1),
+    },
```
Comment (andreaowu) [inline app/frontend/src/features/userPage/UserSection.tsx]: nit: alphabetize

File: app/frontend/src/features/userPage/UserSection.tsx [hunk 10]
```diff
@@ -27,11 +27,9 @@ export default function UserSection({
 }: UserSectionProps) {
   const classes = useStyles();
   return (
-    <Card className={classNames(className, classes.root)}>
-      <CardContent>
-        <Typography variant="h2">{title}</Typography>
-        {children}
-      </CardContent>
-    </Card>
+    <Box className={classNames(className, classes.root)}>
```
Comment (andreaowu) [inline app/frontend/src/features/userPage/UserSection.tsx]: change to div?

**Review by andreaowu — DISMISSED** [review]

File: app/frontend/src/components/UserSummary.tsx (hunk 6 above)
Comment (darrenvong) [reply app/frontend/src/components/UserSummary.tsx]: Surprised Prettier didn't auto format this and left it...

**Comment by darrenvong** [comment]
@Couchers-org/frontend can someone approve this before I get more merge conflicts? 🙃

**Review by lucaslcode — APPROVED** [review]

**Pull request merged**